src/client/js/partials/exposure-scan.js

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/** @type {HTMLDivElement} */
+// @ts-ignore: We guard against a null value by not calling init():
+const exposureScanPartial = document.querySelector("[data-partial='exposureScan']")
+
+if (exposureScanPartial) {
+  init()
+}
+
+async function init () {
+  const urlParams = new URLSearchParams(document.location.search)
+  const emailFromUrl = urlParams.get('email')
+  /** @type {HTMLTemplateElement} */
+  const dataEl = exposureScanPartial.querySelector('#data')
+  dataEl.dataset.email = emailFromUrl
+  const sanitisedEmail = dataEl.dataset.email.trim()
+
+  const res = await fetch('/api/v1/scan/', {
+    headers: {
+      'Content-Type': 'application/json',
+      'x-csrf-token': dataEl.dataset.csrfToken,
+      Accept: 'application/json'
+    },
+    mode: 'same-origin',
+    method: 'POST',
+    body: JSON.stringify({
+      email: sanitisedEmail
+    })
+  })
+
+  if (!res.ok) {
```

Comment:
Yes, good one, and I also updated it to more graciously recover from failures on invalid emails: 8ed8b0ca1

@rhelmer That does mean invalid emails will no longer show up in our logs as errors (though maybe the requests will still be logged?).

Reply:
Yes I think we could figure it out from server logs if someone was abusing this endpoint for instance, there's also generic rate limiting so that should show up in the sentry logs too.